### PR TITLE
Adding reconnect functionality to mongoose

### PR DESCRIPTION
**Pull request by fabianmet**
This is because mongoose does not have a proper reconnect option and the reconnect connection option from mongodb does not cover all failure scenarios.

In this case im afraid the best approach is to retry connection after a failed one.

### CODE REVIEW

File: src/components/mongodb/MongoConnection.ts [hunk 1]
```diff
@@ -20,38 +20,42 @@ export class MongoConnection {
...
         mongoURL: this.config.mongoURL,
       }, 'Mongoose connected');
     });
 
-    mongoose.connection.on('error', (err) => {
-      this.logger.error({
-        error: err,
-      }, 'Mongoose connection error');
+    mongoose.connection.on('reconnected', () => {
+      console.log('MongoDB reconnected!');
```
Comment (ryreitsma) [inline src/components/mongodb/MongoConnection.ts]: Why a console.log here?

File: src/components/mongodb/MongoConnection.ts [hunk 2]
```diff
@@ -20,38 +20,42 @@ export class MongoConnection {
...
     });
 
     mongoose.connection.on('disconnected', () => {
       this.logger.info({}, 'Mongoose disconnected');
+      mongoose.connect(this.config.mongoURL, mongooseOptions);
```
Comment (ryreitsma) [inline src/components/mongodb/MongoConnection.ts]: Won't this lead to a storm of connection attempts if connecting fails? What if the DB is not reachable?

Comment (fabianmet) [reply src/components/mongodb/MongoConnection.ts]: This is intentional and the way mongoose themselves recommend working, i agree so maybe we build in a counter for a max amount of retries with a second delay for instance, ill work on that.

File: src/components/mongodb/MongoConnection.ts (hunk 1 above)
Comment (fabianmet) [reply src/components/mongodb/MongoConnection.ts]: Fixed this

**Pull request merged**